v2/pkg/engine/plan/visitor.go

```diff
@@ -1392,3 +1397,103 @@ func (v *Visitor) resolveFetchDependencies(fetchID int) []resolve.FetchDependenc
 	}
 	return dependencies
 }
+
+func (v *Visitor) buildFetchReasons(fetchID int) []resolve.FetchReason {
+	fields, ok := v.plannerFields[fetchID]
+	if !ok {
+		return nil
+	}
+	dsConfig := v.planners[fetchID].DataSourceConfiguration()
+
+	type typedField struct {
```

Comment:
you could reuse KeyConditionCoordinate and rename it to FieldCoordinate

Reply:
Oh thanks. I have renamed and used it.